test/extended/dr/static_pod_rollouts.go

```diff
@@ -0,0 +1,145 @@
+
+		// select the same node from which the etcd instance was removed to add it back
+		masterNodes := masterNodes(oc)
+		etcdTargetNode := masterNodes[0]
+
+		g.GinkgoT().Log("debugging into the same node to add back the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("node/"+etcdTargetNode.Name, "--", "chroot", "/host", "/bin/bash", "-c", "mv /var/lib/etcd-backup/etcd-pod.yaml /etc/kubernetes/manifests/  && rm -r /var/lib/etcd-backup").Execute()
+		o.Expect(err).ToNot(o.HaveOccurred(), "failed to add back the etcd static pod back to the node")
+
+		// allow some time for the kubelet to act after adding the manifest back
+		time.Sleep(60 * time.Second)
+
+		g.GinkgoT().Log("ensuring that all three etcd instances are available to return the cluster to its original state")
+		etcdPodsCount := getPodCount(oc, etcdNamespace, "app=etcd")
+		o.Expect(etcdPodsCount).To(o.BeNumerically("==", 3))
+
+	})
+
+	g.It("is able to block the rollout of a revision when the quorum is not safe", func(ctx context.Context) {
+		var err error
+
+		g.GinkgoT().Log("ensuring all the three etcd instances are available in the initial state of the cluster")
+		etcdPodsCount := getPodCount(oc, etcdNamespace, "app=etcd")
+		o.Expect(etcdPodsCount).To(o.BeNumerically("==", 3), "three etcd instances should be available in the initial state of the cluster")
+
+		// select a node to remove the etcd instance from
+		masterNodes := masterNodes(oc)
+		etcdTargetNode := masterNodes[0]
+
+		g.GinkgoT().Log("debugging into the node to remove the etcd static pod manifest")
+		err = oc.AsAdmin().Run("debug").Args("node/"+etcdTargetNode.Name, "--", "chroot", "/host", "/bin/bash", "-c", "mkdir /var/lib/etcd-backup && mv /etc/kubernetes/manifests/etcd-pod.yaml /var/lib/etcd-backup").Execute()
+		err = errors.Wrap(err, "failed to remove the etcd static pod from the node")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// allow some time for the kubelet to act after removing the manifest
+		time.Sleep(60 * time.Second)
+
+		g.GinkgoT().Log("ensuring the successful removal of the etcd instance")
+		etcdPodsCount = getPodCount(oc, etcdNamespace, "app=etcd")
+		o.Expect(etcdPodsCount).To(o.BeNumerically("==", 2), "only two etcd instances should be available in the cluster after the removal of one instance")
+
+		g.GinkgoT().Log("ensuring that the removal of the etcd instance marks the corresponding etcd guard as not ready")
+		etcdGuardPod, err := oc.AdminKubeClient().CoreV1().Pods(etcdNamespace).List(ctx, metav1.ListOptions{LabelSelector: "app=guard", FieldSelector: "spec.nodeName=" + etcdTargetNode.Name})
+		o.Expect(err).ToNot(o.HaveOccurred())
+		o.Expect(len(etcdGuardPod.Items)).To(o.BeNumerically("==", 1), "only the correct etcd guard should be retrieved from the node")
+		isEtcdGuardReady := isPodReady(etcdGuardPod.Items[0])
+		o.Expect(isEtcdGuardReady).To(o.BeFalse(), "expected the etcd guard to be marked as not ready after the removal of respective etcd instance")
```

Comment:
it's good that you're testing this, but this is not really related to our quorum guard

Reply:
Would it be best to drop this check?

Reply:
yeah